src/Config/XmlConfigFileLoader.php

```diff
@@ -0,0 +1,220 @@
+            ->create($params);
+    }
+
+    /**
+     * @param string $file
+     * @param array $pshConfigNode
+     */
+    private function setConfigData(string $file, DOMElement $pshConfigNode)
+    {
+        $this->configBuilder->setCommandPaths(
+            $this->extractCommandPaths($file, $pshConfigNode)
+        );
+
+        try {
+            $this->extractPlaceholders(
+                $this->extractNode(self::NODE_PLACEHOLDER, $pshConfigNode)
+            );
+        } catch (\InvalidArgumentException $e) {
+            // nth
+        }
+
+        $this->configBuilder->setTemplates(
+            $this->extractTemplates($file, $pshConfigNode)
+        );
+    }
+
+    /**
+     * @param string $key
+     * @param DOMElement $parent
+     * @return DOMElement[]
+     */
+    private function extractNodes(string $key, DOMElement $parent): array
+    {
+        $nodes = [];
+
+        foreach ($parent->childNodes as $childNode) {
+            if ($childNode instanceof DOMElement && $childNode->localName === $key) {
+                $nodes[] = $childNode;
+            }
+        }
+
+        if (0 === count($nodes)) {
+            return [];
+        }
+
+        return $nodes;
+    }
+
+    /**
+     * @param string $key
+     * @param DOMElement $parent
+     * @return DOMElement
+     */
+    private function extractNode(string $key, DOMElement $parent): DOMElement
+    {
+        $nodes = $this->extractNodes($key, $parent);
+
+        if (count($nodes) === 0) {
+            throw new \InvalidArgumentException('No node found for ' . $key);
+        }
+
+        return $nodes[0];
+    }
+
+    /**
+     * @param string $file
+     * @param $pshConfigNode
+     * @return array
+     */
+    private function extractCommandPaths(string $file, DOMElement $pshConfigNode): array
+    {
+        $pathNodes = $this->extractNodes(self::NODE_PATH, $pshConfigNode);
+
+        return array_map(function (DOMElement $path) use ($file) {
+            return $this->fixPath($this->applicationRootDirectory, $path->nodeValue, $file);
+        }, $pathNodes);
+    }
+
+    /**
+     * @param string $file
+     * @param array $pshConfigNodes
+     * @return array
+     */
+    private function extractTemplates(string $file, DOMElement $pshConfigNodes): array
+    {
+        $templates = $this->extractNodes(self::NODE_TEMPLATE, $pshConfigNodes);
+
+        return array_map(function (DOMElement $template) use ($file) {
+            return [
+                'source' => $this->fixPath(
+                    $this->applicationRootDirectory,
+                    $template->getAttribute(self::NODE_TEMPLATE_SOURCE),
+                    $file
+                ),
+                'destination' => $this->makeAbsolutePath(
+                    $file,
+                    $template->getAttribute(self::NODE_TEMPLATE_DESTINATION)
+                )
+            ];
+        }, $templates);
+    }
+
+    /**
+     * @param DOMElement $placeholder
+     */
```

Comment:
Der php doc block kann von mir aus weg. Steht ja alles schon in der Methoden Signatur drin :-)